File: 283-move-zeroes/move-zeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function(nums) {
    for(let i = 0; i < nums.length-1; i++){
        for(let  j = 0; j < nums.length - i - 1; j++){
            if(nums[j] === 0){
                let tm = nums[j];
                nums[j] = nums[j+1];
                nums[j+1] = tm;
            }
        }
    }
};

File: 283-move-zeroes/move-zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        int i = 0, j = nums.length-1;
        while(i < j){
            if(nums[j] == 0) j--;
            if(nums[i] == 0){
                int k = i;
                while(k < j){
                    int tm = nums[k];
                    nums[k] = nums[k+1];
                    nums[k+1] = tm;
                    k++;
                }
                
            }
            else i++;
        }
    }
}